build failed: error: unknown type name ‘crypto_secretstream_xchacha20poly1305_state’

### Issue and Steps to 
```
git checkout v0.7.3rc2
make clean && make distclean
./configure
make

....

hsmd/hsmd.c: In function ‘create_encrypted_hsm’:
hsmd/hsmd.c:498:2: error: unknown type name ‘crypto_secretstream_xchacha20poly1305_state’
  crypto_secretstream_xchacha20poly1305_state crypto_state;
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:499:12: error: ‘crypto_secretstream_xchacha20poly1305_HEADERBYTES’ undeclared (first use in this function)
  u8 header[crypto_secretstream_xchacha20poly1305_HEADERBYTES];
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:499:12: note: each undeclared identifier is reported only once for each function it appears in
hsmd/hsmd.c:501:36: error: ‘crypto_secretstream_xchacha20poly1305_ABYTES’ undeclared (first use in this function)
  u8 cipher[sizeof(struct secret) + crypto_secretstream_xchacha20poly1305_ABYTES];
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:503:2: error: implicit declaration of function ‘crypto_secretstream_xchacha20poly1305_init_push’ [-Werror=implicit-function-declaration]
  crypto_secretstream_xchacha20poly1305_init_push(&crypto_state, header,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:505:2: error: implicit declaration of function ‘crypto_secretstream_xchacha20poly1305_push’ [-Werror=implicit-function-declaration]
  crypto_secretstream_xchacha20poly1305_push(&crypto_state, cipher,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:501:5: error: unused variable ‘cipher’ [-Werror=unused-variable]
  u8 cipher[sizeof(struct secret) + crypto_secretstream_xchacha20poly1305_ABYTES];
     ^~~~~~
hsmd/hsmd.c:499:5: error: unused variable ‘header’ [-Werror=unused-variable]
  u8 header[crypto_secretstream_xchacha20poly1305_HEADERBYTES];
     ^~~~~~
hsmd/hsmd.c: In function ‘load_hsm’:
hsmd/hsmd.c:636:3: error: unknown type name ‘crypto_secretstream_xchacha20poly1305_state’
   crypto_secretstream_xchacha20poly1305_state crypto_state;
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:637:13: error: ‘crypto_secretstream_xchacha20poly1305_HEADERBYTES’ undeclared (first use in this function)
   u8 header[crypto_secretstream_xchacha20poly1305_HEADERBYTES];
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:639:37: error: ‘crypto_secretstream_xchacha20poly1305_ABYTES’ undeclared (first use in this function)
   u8 cipher[sizeof(struct secret) + crypto_secretstream_xchacha20poly1305_ABYTES];
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:647:7: error: implicit declaration of function ‘crypto_secretstream_xchacha20poly1305_init_pull’ [-Werror=implicit-function-declaration]
   if (crypto_secretstream_xchacha20poly1305_init_pull(&crypto_state, header,
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:651:7: error: implicit declaration of function ‘crypto_secretstream_xchacha20poly1305_pull’ [-Werror=implicit-function-declaration]
   if (crypto_secretstream_xchacha20poly1305_pull(&crypto_state,
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
hsmd/hsmd.c:639:6: error: unused variable ‘cipher’ [-Werror=unused-variable]
   u8 cipher[sizeof(struct secret) + crypto_secretstream_xchacha20poly1305_ABYTES];
      ^~~~~~
hsmd/hsmd.c:637:6: error: unused variable ‘header’ [-Werror=unused-variable]
   u8 header[crypto_secretstream_xchacha20poly1305_HEADERBYTES];
      ^~~~~~
cc1: all warnings being treated as errors
<builtin>: recipe for target 'hsmd/hsmd.o' failed
make: *** [hsmd/hsmd.o] Error 1

```

--- a/configure
+++ b/configure
@@ -238,9 +238,16 @@ link=-lsodium
 code=
 #include <sodium.h>
 #include <stdio.h>
+#include <string.h>
 
 int main(void)
 {
+	crypto_secretstream_xchacha20poly1305_state crypto_state;
+	unsigned char header[crypto_secretstream_xchacha20poly1305_HEADERBYTES];
+	unsigned char data[] = { 1, 2, 3 };
+
+	crypto_secretstream_xchacha20poly1305_init_push(&crypto_state, header,
+	                                                data);
 	printf("%p\n", crypto_aead_chacha20poly1305_ietf_encrypt);
 	printf("%d\n", crypto_aead_chacha20poly1305_ietf_NPUBBYTES);
 	return 0;

